arcs/crowdsourcing_utils.py

```diff
@@ -0,0 +1,336 @@
+    except Exception as e:
+        print("Failed to fetch configuration for %s" % domain)
+        print("Response: %s" % response.content if response else None)
+        print("Exception: %s" % e.message)
+
+
+def gather_columns_types(domain, fxf):
+    """
+    Gather the column headers and their types
+
+    Args:
+        domain (str): A domain cname
+        fxf (str): An identifier for a dataset
+
+    Returns:
+        A list of pairs of containing a column name and a column type
+    """
+    url = 'http://{}/api/views/{}/columns.json'.format(domain, fxf)    
+
+    try:
+        response = requests.get(url)
+    except SSLError:
+        LOGGER.error(
+            "SSLError occurred while gathering column types for {} from domain {}".format(
+                fxf, domain))
+        response = None
+
+    column_types = []
+
+    if response:
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            response_body = []
+        else:
+            response_body = response.json()
+
+            if isinstance(response_body, dict) and "error" in response_body:
+                response_body = []
+
+        def extract(column):
+            return (column.get("name"), column.get("fieldName"), column.get("dataTypeName"))
+
+        column_types = [extract(column) for column in response_body]
+
+    return column_types
+
+
+def convert_to_table(header, rows):
+    """
+    Create an HTML table out of the sample data.
+
+    Args:
+        header (str): The table header
+        rows (List[str]): A list of rows as strings
+
+    Returns:
+        A dataset sample in the form of an HTML <table>
+    """
+    header_html = '<tr><th>{}</th></tr>'.format('</th><th>'.join(header))
+    rows_html = ['<tr><td>{}</td></tr>'.format('</td><td>'.join(row)) for row in rows]
+
+    if rows:
+        table = '<table>{header}\n{rows}</table>'.format(
+            header=header_html, rows='\n'.join(rows_html))
+    else:
+        table = None
+
+    return table
+
+
+def _join_sentences(acc, sentences, max_length):
+    if len(sentences) > 0 and len(acc + " " + sentences[0]) < max_length:
+        return _join_sentences(acc + " " + sentences.pop(0), sentences, max_length)
+    else:
+        return acc.strip()
+
+
+def cleanup_description(desc, nlp=None):
+    """
+    Make a dataset description is readable and not ridiculously long.
+
+    Args:
+        desc (str): A dataset (or other core object) description
+
+    Returns:
+        Returns a trimmed version of the description, containing as many sentences from the
+        description as can fit in a string without exceeding 400 characters
+    """
+    analyzer = nlp or NLP
+    desc = desc.replace("\r", "\n")
+    desc_doc = analyzer(desc) if desc else desc
+    desc_sentences = [s.text.replace("\n", " ").strip() for s in desc_doc.sents] if desc else []
+
+    return _join_sentences("", desc_sentences, 400) if desc else desc
+
+
+def extract_address(cell_contents):
```

Comment:
So much duplicate code...putting this in a shared library is going to be awesome.